Translate: Part of React Reference (#1058)
reactjs/ko.react.dev: src/content/reference/react/index.md

> ## Rules of React {/*rules-of-react*/}

Reviewer: 여기도 'React의 규칙' 으로 번역되면 좋을 것 같습니다

> React에는 패턴을 이해하기 쉽게 하며 고품질의 애플리케이션을 만들 수 있게 하는 관용구(규칙)가 있습니다.

Reviewer: ```suggestion
React에는 패턴을 이해하기 쉽게 하며 고품질의 애플리케이션을 만들 수 있게 하는 일종의 규칙 혹은 모범적인 방식이 있습니다.
```

`idioms`가 직역하면 '관용구' 라는 표현이 맞긴 한데, 여기는 문맥상 의역해서 '모범적인 방식' 등으로 수정하는 편이 더 좋은 것 같습니다.

> * [컴포넌트와 Hook은 순수해야 합니다](/reference/rules/components-and-hooks-must-be-pure) – 코드의 순수성은 코드를 더 쉽게 이해하고 디버그할 수 있도록 하며, React가 바르게 컴포넌트와 훅을 자동으로 최적화할 수 있도록 합니다.

Reviewer: ```suggestion
* [컴포넌트와 Hook은 순수해야 합니다](/reference/rules/components-and-hooks-must-be-pure) – 순수성은 코드를 더 쉽게 이해하고 디버그할 수 있도록 하며, React가 올바르게 컴포넌트와 Hook을 자동으로 최적화할 수 있도록 합니다.
```

> * [React가 컴포넌트와 Hook을 호출하는 방식 ](/reference/rules/react-calls-components-and-hooks) – React는 사용자 경험을 최적화하기 위해 필요할 때마다 컴포넌트와 Hook을 렌더링합니다.

Reviewer: ```suggestion
* [React가 컴포넌트와 Hook을 호출하는 방식](/reference/rules/react-calls-components-and-hooks) – React는 사용자 경험을 최적화하기 위해 필요할 때마다 컴포넌트와 Hook을 렌더링합니다.
```
여기 공백 하나만 지워주세요~
